public/app/features/dashboard-scene/pages/DashboardScenePageStateManager.ts

```diff
@@ -177,6 +178,29 @@ export class DashboardScenePageStateManager extends StateManagerBase<DashboardSc
         return;
       }
 
+      if (config.featureToggles.newDashboardWithFiltersAndGroupBy && Boolean(options.uid)) {
+        // Set preseved filtes from url, only if the
+        const hasFiltersAndGroupBy = dashboardSceneGraph.getFilterAndGroupByVariables(dashboard).length > 0;
+        const preservedQueryParams =
+          store.get('grafana.dashboard.preservedUrlFiltersState') &&
+          JSON.parse(store.get('grafana.dashboard.preservedUrlFiltersState'));
+        if (hasFiltersAndGroupBy && preservedQueryParams) {
+          const currentQueryParams = locationService.getLocation().search;
+
+          // TODO: deduplicate params with the same value, i.e. for a case when someone navigates back to the same dashboard
+          let nextQueryParams = currentQueryParams;
+          if (currentQueryParams) {
+            nextQueryParams += '&' + urlUtil.renderUrl('', preservedQueryParams).slice(1);
+          } else {
+            nextQueryParams = urlUtil.renderUrl('', preservedQueryParams);
+          }
+          console.log('setting preserved filters from url', nextQueryParams);
+          locationService.replace({
+            search: nextQueryParams,
+          });
+        }
+      }
```

Comment:
I would skip the `dashboardSceneGraph.getFilterAndGroupByVariables(dashboard).length > 0;` check here and just take all the url keys found in the `preserveUrlState` localStorage string, and for each URL key (`var-<name>`) look up if the variable exists in the dashboard and if not exclude it. (always include time range from & to)